internal/sdkv2provider/schema_cloudflare_access_application.go

```diff
@@ -54,13 +54,53 @@ func resourceCloudflareAccessApplicationSchema() map[string]*schema.Schema {
 				return oldValue == newValue
 			},
 		},
+		"domain_type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringInSlice([]string{"public", "private"}, false),
+			Description:  fmt.Sprintf("The type of the primary domain. %s", renderAvailableDocumentationValuesStringSlice([]string{"public", "private"})),
+			DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+				appType := d.Get("type").(string)
+				// Suppress the diff if it's an app type that doesn't need a `domain` value.
+				if appType == "infrastructure" {
+					return true
+				}
+
+				return oldValue == newValue
+			},
+		},
+		"destinations": {
```

Comment:
Should this be named `destination` (not plural) instead? It'll be used like this:

```terraform
destinations {
  uri = "example.com"
}
destinations {
  type = "private"
  uri = "10.116.0.4"
}
```

Reply:
I saw the [Attributes as Blocks](https://developer.hashicorp.com/terraform/language/attr-as-blocks) page but it's kind of confusing to me.

Reply:
`destinations` makes sense. in v5, this makes a lot more sense as a collection of objects - https://github.com/cloudflare/terraform-provider-cloudflare/blob/next/internal/services/zero_trust_access_application/schema.go#L225-L244